components/style/properties/shorthand/font.mako.rs

```diff
@@ -97,14 +97,25 @@
     // This may be a bit off, unsure, possibly needs changes
     impl<'a> ToCss for LonghandsToSerialize<'a>  {
         fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-            self.font_style.to_css(dest)?;
-            dest.write_str(" ")?;
-            self.font_variant.to_css(dest)?;
-            dest.write_str(" ")?;
-            self.font_weight.to_css(dest)?;
-            dest.write_str(" ")?;
-            self.font_stretch.to_css(dest)?;
+
+    % if product == "gecko":
+        % for name in "size_adjust kerning variant_caps variant_position".split():
+            if self.font_${name}.is_some() {
```

Comment:
It appears that all are single keyword properties or have `get_initial_specified_value()` so this change is all that's needed here. Thanks.